### PR TITLE
[Enhancement] Make schema changed table not to be moved to trash

### LINKED ISSUE #13651: [Enhancement] Can remove old tablet files directly, instead of move into trash when drop/add index
## Enhancement
since drop/add index cannot recover, can remove old tablet files directly, instead of moving into trash.

### PR DESCRIPTION
**Pull request by zaorangyang**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #13651

## Problem Summary(Required) ：

When a table schema change is complete, the old table is moved to the trash directory, which takes up a lot of disk space. This PR fixes this issue.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/catalog/TabletInvertedIndex.java [hunk 1]
```diff
@@ -89,7 +89,7 @@ public class TabletInvertedIndex {
     // replica id -> tablet id
     private Map<Long, Long> replicaToTabletMap = Maps.newHashMap();
 
-    private Set<Long> truncatedTablets = Sets.newHashSet();
+    private Set<Long> forceDeleteTablets = Sets.newHashSet();
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/catalog/TabletInvertedIndex.java:92]: How about a table add a materialized view?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 11 / 11 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/leader/ReportHandler.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/TabletInvertedIndex.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeJobV2.java|4|4|100.00%|[]|

**Comment by wanpengfei-git** [comment]
@Mergify rebase

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 11 / 11 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/leader/ReportHandler.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/TabletInvertedIndex.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeJobV2.java|4|4|100.00%|[]|

**Review by chaoyli — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**